ui/components/multichain/network-list-menu/network-list-menu.tsx

```diff
@@ -291,78 +332,138 @@ export const NetworkListMenu = ({ onClose }: { onClose: () => void }) => {
+
+    const onDelete = () => {
+      dispatch(toggleNetworkMenu());
+      dispatch(
+        showModal({
+          name: 'CONFIRM_DELETE_NETWORK',
+          target: convertCaipToHexChainId(network.chainId),
+          onConfirm: () => undefined,
+        }),
+      );
+    };
+
+    const onEdit = () => {
+      dispatch(
+        setEditedNetwork({
+          chainId: network.chainId,
+          nickname: network.name,
+        }),
+      );
+      setActionMode(ACTION_MODE.ADD_EDIT);
+    };
+
+    const onRpcConfigEdit = () => {
+      setActionMode(ACTION_MODE.SELECT_RPC);
+      dispatch(setEditedNetwork({ chainId: network.chainId }));
+    };
+
+    const iconSrc = network.isEvm
+      ? CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP[
+          convertCaipToHexChainId(
+            network.chainId,
+          ) as keyof typeof CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP
+        ]
+      : MULTICHAIN_TOKEN_IMAGE_MAP[network.chainId];
 
     return (
       <NetworkListItem
-        name={network.name}
-        iconSrc={
-          CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP[
-            network.chainId as keyof typeof CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP
-          ]
-        }
-        iconSize={AvatarNetworkSize.Sm}
-        rpcEndpoint={
-          showMultiRpcSelectors
-            ? network.rpcEndpoints[network.defaultRpcEndpointIndex]
-            : undefined
-        }
         key={network.chainId}
         chainId={network.chainId}
+        name={network.name}
+        iconSrc={iconSrc}
+        iconSize={AvatarNetworkSize.Sm}
         selected={isCurrentNetwork && !focusSearch}
         focus={isCurrentNetwork && !focusSearch}
-        onClick={() => {
-          handleNetworkChange(network);
-        }}
-        onDeleteClick={
-          canDeleteNetwork
-            ? () => {
-                dispatch(toggleNetworkMenu());
-                dispatch(
-                  showModal({
-                    name: 'CONFIRM_DELETE_NETWORK',
-                    target: network.chainId,
-                    onConfirm: () => undefined,
-                  }),
-                );
-              }
+        rpcEndpoint={
+          hasMultiRpcOptions
+            ? getDefaultRpcEndpointByChainId(network.chainId)
             : undefined
         }
-        onEditClick={() => {
-          dispatch(
-            setEditedNetwork({
-              chainId: network.chainId,
-              nickname: network.name,
-            }),
-          );
-          setActionMode(ACTION_MODES.ADD_EDIT);
-        }}
-        onRpcEndpointClick={() => {
-          setActionMode(ACTION_MODES.SELECT_RPC);
-          dispatch(setEditedNetwork({ chainId: network.chainId }));
+        onClick={async () => {
+          await handleNetworkChange(network.chainId);
         }}
+        onDeleteClick={isDeletable ? () => onDelete() : undefined}
+        onEditClick={isEditable ? () => onEdit() : undefined}
+        onRpcEndpointClick={network.isEvm ? onRpcConfigEdit : undefined}
```

Comment:
```suggestion
        onRpcEndpointClick={network.isEvm ? () => onRpcConfigEdit() : undefined}
```

Reply:
Done in https://github.com/MetaMask/metamask-extension/commit/119a7a4fb2fd2a22a7e4f665f5ebcbfec9c1e1ef